### PR TITLE
Tool time

### PR DESCRIPTION
**Pull request by nayrpg**

### CODE REVIEW

File: Assets/Scripts/Items/InventoryManager.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	// Update is called once per frame
+	void Update ()
+    {
+
+        // Bit shift the index of the layer (8) to get a bit mask
+        int layerMask = 0 << 2;
+
+        // This would cast rays only against colliders in layer 8.
+        // But instead we want to collide against everything except layer 8. The ~ operator does this, it inverts a bitmask.
+        layerMask = ~layerMask;
```
Comment (Feesh7) [inline Assets/Scripts/Items/InventoryManager.cs]: 🗾

File: Assets/Scripts/Items/InventoryManager.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        shipped_item.ctx = this;
+        shipped_item.Start();
+        shipped_item.menu_form = image;
+        Debug.Log("Adding item");
+        if (UseItem.isUseItem(shipped_item)) {
+            Debug.Log("Use item");
+            actionSlots.AddUseItem((UseItem) shipped_item);
+        }
+        if (AbilityItem.isAbilityItem(shipped_item)) {
+            Debug.Log("Ability item");
```
Comment (Feesh7) [inline Assets/Scripts/Items/InventoryManager.cs]: 👺

File: Assets/Scripts/Items/InventoryManager.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    void AddItemToInventory (ItemRequest request)
+    {
+        Item shipped_item = amazon.RequestItem(request);
+
+        shipped_item.ctx = this;
+        shipped_item.Start();
+        shipped_item.menu_form = image;
+        Debug.Log("Adding item");
+        if (UseItem.isUseItem(shipped_item)) {
+            Debug.Log("Use item");
```
Comment (Feesh7) [inline Assets/Scripts/Items/InventoryManager.cs]: 👺

File: Assets/Scripts/SceneLoader.cs [hunk 4]
```diff
@@ -23,14 +23,8 @@ IEnumerator LoadAsyncNewScene(string scene)
...
-        GameObject[] playerObjects = GameObject.FindGameObjectsWithTag("Player");
-        foreach (GameObject player in playerObjects) {
-            if ( player == gameObject || player.transform.root != transform) {
-                transform.parent = null;
-                // Move the GameObject (you attach this in the Inspector) to the newly loaded Scene
-                SceneManager.MoveGameObjectToScene(player, newScene);
-            }
-        }
+        // Move the GameObject (you attach this in the Inspector) to the newly loaded Scene
+        SceneManager.MoveGameObjectToScene(gameObject, newScene);
```
Comment (Feesh7) [inline Assets/Scripts/SceneLoader.cs]: 👺

File: Assets/Scripts/InputManager.cs [hunk 5]
```diff
@@ -85,6 +142,7 @@ private void UpdateInputs()
         _input_horizontal_axis = Input.GetAxisRaw("Horizontal");
         _input_scroll_axis = Input.GetAxis("Mouse ScrollWheel");
         MouseUpdate();
+        Debug.DrawRay(transform.position, transform.up * Input.GetAxisRaw("Ability Slot 2"), Color.green);
```
[the remark this answers is not in the document]
Comment (nayrpg) [reply Assets/Scripts/InputManager.cs]: Done

File: Assets/Scripts/Inventory.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    public bool enable_logs = true;
+
+    public Dictionary<string, T> contents;
+    public uint numSlots = 1;
+
+    // Use this for initialization
+    public Inventory()
+    {
+        contents = new Dictionary<string, T>();
+        Debug.Log(contents.Values);
```
[the remark this answers is not in the document]
Comment (nayrpg) [reply Assets/Scripts/Inventory.cs]: Done

File: Assets/Scripts/ItemCatalogue.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        items = new Dictionary<string, ItemCatalogueEntry>();
+        items.Add(MagnetBoots.item_name, new ItemCatalogueEntry(typeof(MagnetBoots), typeof(AbilityItem), "", ""));
+    }
+    public Item RequestItem (ItemRequest req)
+    {
+        Item requested_item = Activator.CreateInstance(items[req.item_name].type) as Item;
+
+        Debug.Log(requested_item.GetName());
+        Debug.Log(requested_item.type);
```
[the remark this answers is not in the document]
Comment (nayrpg) [reply Assets/Scripts/ItemCatalogue.cs]: Done

File: Assets/Scripts/Items/InventoryManager.cs (hunk 1 above)
Comment (nayrpg) [reply Assets/Scripts/Items/InventoryManager.cs]: Done

File: Assets/Scripts/Items/InventoryManager.cs (hunk 2 above)
Comment (nayrpg) [reply Assets/Scripts/Items/InventoryManager.cs]: Done

File: Assets/Scripts/Items/InventoryManager.cs (hunk 3 above)
Comment (nayrpg) [reply Assets/Scripts/Items/InventoryManager.cs]: Done

File: Assets/Scripts/SceneLoader.cs (hunk 4 above)
Comment (nayrpg) [reply Assets/Scripts/SceneLoader.cs]: will be done in the future.

**Pull request merged**